src/sentry/api/endpoints/group_integration_details.py

```diff
@@ -168,56 +172,66 @@ def put(self, request: Request, group, integration_id) -> Response:
+            action_type=ProjectManagementActionType.LINK_EXTERNAL_ISSUE,
+            integration=installation.model,
+        ).capture() as lifecycle:
+            try:
+                data = installation.get_issue(external_issue_id, data=request.data)
+            except IntegrationFormError as exc:
+                lifecycle.record_halt(exc)
+                return Response(exc.field_errors, status=400)
+            except IntegrationError as e:
+                lifecycle.record_halt(e)
+                return Response({"non_field_errors": [str(e)]}, status=400)
+
+            defaults = {
+                "title": data.get("title"),
+                "description": data.get("description"),
+                "metadata": data.get("metadata"),
+            }
 
-        if created:
-            integration_issue_linked.send_robust(
-                integration=integration,
-                organization=group.project.organization,
-                user=request.user,
-                sender=self.__class__,
+            external_issue_key = installation.make_external_key(data)
+            external_issue, created = ExternalIssue.objects.get_or_create(
+                organization_id=organization_id,
+                integration_id=integration.id,
+                key=external_issue_key,
+                defaults=defaults,
             )
-        else:
-            external_issue.update(**defaults)
-
-        installation.store_issue_last_defaults(group.project, request.user, request.data)
-        try:
-            installation.after_link_issue(external_issue, data=request.data)
-        except IntegrationFormError as exc:
-            return Response(exc.field_errors, status=400)
-        except IntegrationError as e:
-            return Response({"non_field_errors": [str(e)]}, status=400)
 
-        try:
-            with transaction.atomic(router.db_for_write(GroupLink)):
-                GroupLink.objects.create(
-                    group_id=group.id,
-                    project_id=group.project_id,
-                    linked_type=GroupLink.LinkedType.issue,
-                    linked_id=external_issue.id,
-                    relationship=GroupLink.Relationship.references,
+            if created:
+                integration_issue_linked.send_robust(
+                    integration=integration,
+                    organization=group.project.organization,
+                    user=request.user,
+                    sender=self.__class__,
                 )
-        except IntegrityError:
-            return Response({"non_field_errors": ["That issue is already linked"]}, status=400)
+            else:
+                external_issue.update(**defaults)
+
+            installation.store_issue_last_defaults(group.project, request.user, request.data)
+            try:
+                installation.after_link_issue(external_issue, data=request.data)
+            except IntegrationFormError as exc:
+                lifecycle.record_halt(exc)
```

Comment:
are these errors user config errors?

Reply:
Yeah, typically an `IntegrationFormError` is an incorrectly configured Jira ticket creation config.